### PR TITLE
Rename Request to HttpRequest

### PR DESCRIPTION
**Pull request by pakrym**
And move to Azure.Core.Http namespace.

Contributes to #6831

### CODE REVIEW

File: sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts [hunk 1]
```diff
@@ -427,7 +427,7 @@ function generateOperation(w: IndentWriter, serviceModel: IServiceModel, group:
                         // an unimplemented method on the partial class
                         w.line(`throw ${valueName}.CreateException(${responseName});`);
                     } else {
-                        w.line(`throw new Azure.RequestFailedException(${responseName});`);
+                        w.line(`throw new Azure.Core.Http.HttpRequestFailedException(${responseName});`);
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts]: Did you mean to include this here?

**Review by tg-msft — APPROVED** [review]
I think there's a small issue with the generator, but it's otherwise good.

File: sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts (hunk 1 above)
Comment (pakrym) [reply sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts]: Good catch, nope.

**Review by schaabs — COMMENTED** [review]
Is there any concern that there is an HttpRequest class in System.Web which this might be confused with? I could make finding documentation for this type more tricky. Also, while I don't think the two would have to be used together very often, there is a case in Azure.Identity where I have to use both Azure.Core's Request and System.Web HttpReqeust. But if it's very uncommon it can be handled by fully qualifying the type names.

**Comment by pakrym** [comment]
> Is there any concern that there is an HttpRequest class in System.Web which this might be confused with 

Yes, it is a concern, VS intellisense might be confused when trying to add using statements. I'll put this PR on hold.

**Comment by pakrym** [comment]
I've reverted the class name change leaving only the namespace.

**Pull request merged**